packages/frontend-2/lib/projects/composables/projectManagement.ts

```diff
@@ -133,6 +137,37 @@ export function useCreateProject() {
               },
               { fieldNameWhitelist: ['admin'] }
             )
+
+            // New cache update for Workspace.projects
+            if (input.workspaceId) {
+              const workspaceCacheId = getCacheId('Workspace', input.workspaceId)
+
+              modifyObjectFields<
+                { filter?: { search?: string } },
+                { items: Reference[]; totalCount: number }
+              >(cache, workspaceCacheId, (fieldName, variables, value, details) => {
+                if (fieldName !== 'projects') return
+                if (variables?.filter?.search?.length) return
+
+                const newVal = { ...value }
+                if (Array.isArray(value.items)) {
+                  newVal.items = [details.ref('Project', newProject.id), ...value.items]
+                } else {
+                  newVal.items = [details.ref('Project', newProject.id)]
+                }
+                newVal.totalCount = (value.totalCount || 0) + 1
```

Comment:
There's a reason all of the fields of `value` are optional - you don't know which if any of those fields are actually going to be requested and in the cache. If `totalCount` is undefined, meaning - not requested, you can't just default to 0. Just because we didn't ask for a totalCount from the API (the list may actually have a 100 workspaces), doesn't mean that we should now show it has 1 on the frontend.

The correct way to handle this is to only update totalCount if it exists in the cache. If it doesn't - leave it as is.

```
const newTotalCount = isUndefined(value?.totalCount) ? undefined : Math.max(0, (value?.totalCount || 0) - 1)
return {
  ...value,
  ...(isUndefined(newTotalCount) ? {} : {totalCount: newTotalCount})
}
```

Same goes for all other fields like `items`